app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -195,35 +228,11 @@ function ContainerTopologyCtrl($scope, $http, $interval, $location, topologyServ
   };
 
   this.getIcon = function getIcon(d) {
-    switch (d.item.kind) {
-      case 'Container':
-        return '\uF1B2'; // fa-cube
-      case "ContainerNode":
-        return '\uE621';  // pficon-container-node
-      case "ContainerRoute":
-        return '\uE625'; // pficon-route
-      case "ContainerService":
-        return '\uE61E'; // pficon-service
-      case "Vm":
-        return '\uE90f'; // pficon-virtual-machine
-      case "Host":
-        return '\uE600'; // pficon-screen
-      case "ContainerGroup":
-        return '\uF1B3'; // fa-cubes
-      case "ContainerReplicator":
-        return '\uE624'; // pficon-replicator
-      case "ContainerManager":
-        switch (d.item.display_kind) {
-          case "Kubernetes":
-            return '\uE627'; // pficon-kubernetes
-          case "Openshift":
-          case "OpenshiftEnterprise":
-            return '\uE626';  // pficon-openshift
-          case "Atomic":
-            return '\uE62c'; // vendor-atomic
-          case "AtomicEnterprise":
-            return '\uE62d'; //vendor-atomic-enterprise
-        }
+    switch(d.item.kind) {
+      case 'ContainerManager':
+        return icons[d.item.display_kind];
```

Comment:
how are the icons handled in the system in other places then? they are not part of the model afaik. also, please keep in mind that a large part of the data passed to topology is because the topology widget asks for this data, and it is reused here in the controller for additional functionaulity.